Loop/Managers/DeviceDataManager.swift

```diff
@@ -1443,30 +1442,140 @@ extension DeviceDataManager {
+        
+        let absorptionTime = absorptionTime ?? carbStore.defaultAbsorptionTimes.medium
+        if absorptionTime < LoopConstants.minCarbAbsorptionTime || absorptionTime > LoopConstants.maxCarbAbsorptionTime {
+            throw CarbActionError.invalidAbsorptionTime(absorptionTime)
+        }
+        
+        guard amountInGrams > 0.0 else {
+            throw CarbActionError.invalidCarbs
+        }
+        
+        guard amountInGrams <= LoopConstants.maxCarbEntryQuantity.doubleValue(for: .gram()) else {
+            throw CarbActionError.exceedsMaxCarbs
+        }
+        
+        if let startDate = startDate {
+            let maxStartDate = Date().addingTimeInterval(LoopConstants.maxCarbEntryFutureTime)
+            let minStartDate = Date().addingTimeInterval(LoopConstants.maxCarbEntryPastTime)
+            guard startDate <= maxStartDate  && startDate >= minStartDate else {
+                throw CarbActionError.invalidStartDate(startDate)
+            }
+        }
+        
+        let quantity = HKQuantity(unit: .gram(), doubleValue: amountInGrams)
+        let candidateCarbEntry = NewCarbEntry(quantity: quantity, startDate: startDate ?? Date(), foodType: foodType, absorptionTime: absorptionTime)
         
         let _ = try await addRemoteCarbEntry(candidateCarbEntry)
+        await NotificationManager.sendRemoteCarbEntryNotification(amountInGrams: candidateCarbEntry.quantity.doubleValue(for: .gram()))
         await remoteDataServicesManager.triggerUpload(for: .carb)
     }
     
+    enum CarbActionError: LocalizedError {
+        
+        case invalidAbsorptionTime(TimeInterval)
+        case invalidStartDate(Date)
+        case exceedsMaxCarbs
+        case invalidCarbs
+        
+        var errorDescription: String? {
+            switch  self {
+            case .exceedsMaxCarbs:
+                return NSLocalizedString("Exceeds maximum allowed carbs", comment: "Remote command error description: carbs exceed maximum amount.")
+            case .invalidCarbs:
+                return NSLocalizedString("Invalid carb amount", comment: "Remote command error description: invalid carb amount.")
+            case .invalidAbsorptionTime(let absorptionTime):
+                return String(format: NSLocalizedString("Invalid absorption time: %d hours", comment: "Remote command error description: invalid absorption time."), absorptionTime.hours)
+            case .invalidStartDate(let startDate):
+                return String(format: NSLocalizedString("Start time is out of range: %@", comment: "Remote command error description: invalid start time is out of range."), Self.dateFormatter.string(from: startDate))
+            }
+        }
+        
+        static var dateFormatter: DateFormatter = {
+            let formatter = DateFormatter()
+            formatter.timeStyle = .medium
+            return formatter
+        }()
+    }
+    
```

Comment:
REMOTE 2.0: WILL REMOVE PRE-MERGE

Referring to `handleRemoteAutobolus` and `handleRemoteClosedLoop`.

Reply:
Remote settings updates might make sense (overrides are already a form of remote settings updating), but we should be letting the Loop user know they are happening.

Reply:
Thanks - I'll look into that later on with the Remote 2.0 work. I've since removed the remote 2.0 work here.